### PR TITLE
Attestation procedure corrections

### PR DESCRIPTION
**Pull request by beneichorst**
Correcting OpenSSL commands to provide for proper validation examples using OpenSSL.

### CODE REVIEW

**Comment by beneichorst** [comment]
Ths PR is a short-term fix -- we may revisit documentation for PIV attestation in the near future

File: doc/Attestation.adoc [hunk 1]
```diff
@@ -7,32 +7,31 @@ A high level description of the thinking and how this can be used can be found
 at https://developers.yubico.com/PIV/Introduction/PIV_attestation.html
 
 === Usage
-Attestation works through a special key slot called “f9” this comes
-pre-loaded from factory with a key and cert signed by Yubico, but can be
-overwritten.
-After a key has been generated in a normal slot it can be attested by this
-special key, this can be realised by using the yubico-piv-tool action attest:
+The only purposes of PIV attestation is to show that a certain asymmetric key has been generated on device and not imported.
```
Comment (happyfuncode) [inline doc/Attestation.adoc:10]: I'd remove this editorial comment until we resolve that discussion internally.

File: doc/Attestation.adoc [hunk 2]
```diff
@@ -7,32 +7,31 @@ A high level description of the thinking and how this can be used can be found
...
+
+Attestation works through a special key slot called “F9” that comes
+pre-loaded from the factory with a key and certificate signed by Yubico's root PIV CA. While not recommended for most use cases, this certificate can be overwritten. Note that, if overwritten, the original signing key and certificate in slot F9 cannot be restored. The attestation key and certificate will not be cleared out by a reset of the device.
```
Comment (happyfuncode) [inline doc/Attestation.adoc:13]: I'd put the advanced overwriting f9 use case in a separate document, a footnote, or at least an "advanced" section at the bottom.

File: doc/Attestation.adoc [hunk 3]
```diff
@@ -7,32 +7,31 @@ A high level description of the thinking and how this can be used can be found
...
+After a key has been generated in another slot, the fact that key material was generated on the YubiKey can be attested using the yubico-piv-tool action attest:
```
Comment (happyfuncode) [inline doc/Attestation.adoc:14]: same, let's leave the editorial and repetitive comment out until later.

File: doc/Attestation.adoc [hunk 4]
```diff
@@ -7,32 +7,31 @@ A high level description of the thinking and how this can be used can be found
...
+  
 Now we're ready to verify the attestation:
 
-  $ yubico-piv-tool --action=attest --slot=9a > attestation.pem
-  $ openssl verify -CAfile certs.pem attestation.pem
+  $ yubico-piv-tool --action=attest --slot=9a > Slot9Aattestation.pem
+  $ openssl verify -CAfile TrustedCAcerts.pem -untrusted SlotF9Intermediate.pem Slot9Aattestation.pem
   attestation.pem: OK
 
 [NOTE]
```
Comment (happyfuncode) [inline doc/Attestation.adoc:37]: I think the changes on lines 20-34 are perfect.

I'd add a line about openssl verify being a fine way to test, but suggesting it not be used in production or similar.

File: doc/Attestation.adoc (hunk 1 above)
Comment (beneichorst) [reply doc/Attestation.adoc:10]: This is taken with slight modification from the PIV attestation documentation linked earlier in this doc here: https://developers.yubico.com/PIV/Introduction/PIV_attestation.html

We can remove this for now, but let's keep a mental note here that we'll need to rectify it with our other documentation for PIV attestation here, as well as the YESDK documentation.

Comment (beneichorst) [reply doc/Attestation.adoc]: Removed - we will iron out additional changes to this page later.

File: doc/Attestation.adoc (hunk 4 above)
Comment (beneichorst) [reply doc/Attestation.adoc:37]: Note added for verify being non-prod

**Review by qpernil — APPROVED** [review]

**Review by aveenismail — APPROVED** [review]

**Pull request merged**